predicators/approaches/bridge_policy_approach.py

```diff
@@ -410,48 +473,43 @@ def _solve(self,
                task: Task,
                timeout: int,
                train_or_test: str = "test") -> Callable[[State], Action]:
-        if not self._maple_initialized:
-            self.mapleq = MapleQApproach(self._get_current_predicates(),
-                                         self._initial_options, self._types,
-                                         self._action_space, self._train_tasks)
-            self._maple_initialized = True
-            self._init_nsrts()
-
         # Start by planning. Note that we cannot start with the bridge policy
         # because the bridge policy takes as input the last failed NSRT.
-        current_control = "planner"
+        self._current_control = "planner"
         option_policy = self._get_option_policy_by_planning(task, timeout)
-        current_policy = utils.option_policy_to_policy(
+        self._current_policy = utils.option_policy_to_policy(
             option_policy,
             max_option_steps=CFG.max_num_steps_option_rollout,
             raise_error_on_repeated_state=True,
         )
-        all_failed_options: List[_Option] = []
-
-        # Prevent infinite loops by detecting if the bridge policy is called
-        # twice with the same state.
-        last_bridge_policy_state = DefaultState
+        if not self._maple_initialized:
+            self.mapleq = MapleQApproach(self._get_current_predicates(),
+                                         self._initial_options, self._types,
+                                         self._action_space, self._train_tasks)
+            self._maple_initialized = True
+            self._init_nsrts()
 
         def _policy(s: State) -> Action:
-            nonlocal current_control, current_policy, last_bridge_policy_state
-
             # Normal execution. Either keep executing the current option, or
             # switch to the next option if it has terminated.
             try:
-                action = current_policy(s)
+                action = self._current_policy(s)
+                if train_or_test == "train":
+                    self._policy_logs.append(self._current_control)
                 return action
-            except OptionExecutionFailure as e:
-                failed_option = e.info["last_failed_option"]
-                if failed_option is not None:
-                    all_failed_options.append(failed_option)
+            except utils.OptionExecutionFailure:
+                print("failed", self._current_control)
```

Comment:
in general, we've moved away from using `print` statements in the codebase. Instead, we use `logging.debug` or `logging.info` (you can grep for these and see how they're used elsewhere)